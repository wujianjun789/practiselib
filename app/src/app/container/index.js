/**
 * Created by a on 2017/6/29.
 */
import React, { Component } from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux'
import PropTypes from 'prop-types';
import '../../../public/styles/app.less';
import Card from './Card';
import UserCenter from '../../common/containers/UserCenter';
import Overlayer from '../../common/containers/Overlayer';
import {loginHandler} from '../../api/login'
import {getModule} from '../action/index'
/**
 * @param {String} title required
 * @param {String} name  required
 * @param {Array}  items required
 */
export class App extends Component{
    constructor(props){
        super(props);
        this.loginFail = this.loginFail.bind(this);
        this.loginSuccess = this.loginSuccess.bind(this);
    }

    componentWillMount(){
        console.log(localStorage)
        /*if(sessionStorage.sessionID==0||sessionStorage.sessionID==null||sessionStorage.sessionID==""){
            
            this.props.router.push('/login')
            
        }else{

            this.props.actions.loginHandler(sessionStorage.username, sessionStorage.password, this.loginSuccess, this.loginFail);

        }*/
        // if(this.props.userCenter.islogin!=1){
        //     this.props.router.push('/login')
        // }
        const {actions} = this.props;
        actions && actions.getModule();
    }

    componentDidMount() {
    }

    componentWillReceiveProps(nextProps) {



    }

    loginFail(){
        this.setState({style:{visibility: 'visible'}})
    }

    loginSuccess(){
        this.props.router.push('/')
    }

    render(){
        const {title="StarRiver", name="智慧路灯管理系统", items} = this.props;
        return (
            <div className="app">
                <div className="header clearfix">
                    <div className="header-left clearfix">
                        <span className="icon icon-logo"></span>
                        <span className="tit">{title}</span>
                    </div>
                    <UserCenter router={this.props.router}/>
                    <div className="header-right clearfix">
                        <span className="name">{name}</span>
                    </div>
                </div>
                <div className="cont">
                    <ul className="clearfix">
                        {items.map((item, index)=><li key={item.key}><Card _key={item.key} title={item.title} link={item.link}/></li>)}
                    </ul>
                </div>
                <Overlayer />
            </div>
        )
    }
}

const mapStateToProps = (state, ownProps) => {
    let app = state.app;
    return {
        title: app.title,
        name: app.name,
        items: app.items,
        userCenter:state.userCenter
    }
}

const mapDispatchToProps = (dispatch, ownProps) => {
    return {
        actions: bindActionCreators({
            getModule: getModule,
            loginHandler
        }, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(App);

App.propTypes = {
    title: PropTypes.string,
    name: PropTypes.string,
    items: PropTypes.arrayOf(
        PropTypes.shape({
            key: PropTypes.string.isRequired,
            title: PropTypes.string.isRequired,
            link: PropTypes.string.isRequired
        })
    ).isRequired
}