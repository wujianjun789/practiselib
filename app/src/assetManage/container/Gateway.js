/**
 * Created by m on 2018/2/9.
 */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import Content from '../../components/Content';

import { getModelData, getModelProps, getModelDefaultsValues, getModelDefaults, deleteModalTypes, addModalTypes }
  from '../../data/assetModels';
import { addModalTypesById } from '../../api/asset';
import Immutable from 'immutable';

import { injectIntl } from 'react-intl';
import { intlFormat } from '../../util/index';
import ConfirmPopup from '../../components/ConfirmPopup';
import TypeEditPopup from '../components/TypeEditPopup';
import { overlayerShow, overlayerHide } from '../../common/actions/overlayer';
import { getModelTypeByModel, updateModelTypeByModel } from '../../api/asset';
import { addNotify } from '../../common/actions/notifyPopup';
import NotifyPopup from '../../common/containers/NotifyPopup';

export class Gateway extends Component {
  constructor(props) {
    super(props);
    this.state = {
      model: 'ssgw',
      keyField: 'name',   //主键在数据表的Id属性中，即设备类型的id

      assetPropertyList: [   //设备属性列表
        // { id: '111', name: '三思LC', description: '三思单灯控制器', unit: '0000', accuracy: '0000' },
        // { id: '123', name: '华为LC_NBLot', description: '华为单灯控制器', unit: '0000', accuracy: '0000' },
      ],
      assetTypeList: [     //设备型号列表
        { name: '', description: '' }
        // { name: '华为单灯', description: '华为单灯控制器', power: '0000', life: '0000', manufacture: '0000' },
        // {name: '', detail: '', power: '', serviceLife: '', manufacture: '' },
      ],
    };


    this.assetTypeColumns = [  //ssslc设备型号表头定义types, name, description, power, life, manufacture
      { field: 'name', title: this.formatIntl('equipment.name')},
      { field: 'description', title: this.formatIntl('equipment.description') },
      { field: 'power', title: this.formatIntl('equipment.power') },
      { field: 'life', title: this.formatIntl('equipment.life') },
      { field: 'manufacture', title: this.formatIntl('equipment.manufacturer') },
    ];
    this.formatIntl = this.formatIntl.bind(this);
    this.rowDelete = this.rowDelete.bind(this);
    this.rowEdit = this.rowEdit.bind(this);
    this.rowAdd = this.rowAdd.bind(this);
    this.getModelType = this.getModelType.bind(this);
    this.updateModelType = this.updateModelType.bind(this);
  }

  componentDidMount() {
    let { model, assetPropertyList } = this.state;
    this.mounted = true;
    getModelData(() => {
      this.mounted && this.setState({ assetPropertyList: getModelProps(model) }, () => {
      });
    });
    this.getModelType();
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  formatIntl(formatId) {
    const { intl } = this.props;
    return intl ? intl.formatMessage({ id: formatId }) : null;
  }

  getModelType() {
    let { assetTypeList, model } = this.state;
    getModelTypeByModel(model, (data) => {
      let types = data[0].types;
      this.mounted && this.updateModelType(types);
    });
  }

  updateModelType(data) {
    let dataItem = { name: '', description: '' }
    data.push(dataItem);
    this.setState({ assetTypeList: data })
  }

  deleteModalTypeById(id) {
    const { assetTypeList, model } = this.state;
    let data = assetTypeList;
    let dataleft = _.remove(data, function (n) {
      return n.name === id;
    });
    data.pop();
    updateModelTypeByModel(model, data, (data) => {
      this.mounted && this.getModelType();
    })
  }

  rowDelete(id) { //id为型号名称
    const { data } = this.state;
    const { actions } = this.props;
    let curId = id; //当前要删除的型号的Id
    actions.overlayerShow(<ConfirmPopup iconClass="icon_popup_delete"
      tips={this.formatIntl('equipment.isDeleteDevice')} cancel={() => { actions.overlayerHide(); }}
      confirm={() => {
        this.deleteModalTypeById(curId);
        actions.overlayerHide();
      }} />);
  }

  rowEdit(id) { //id为选中的设备型号的name，
    let curId = id;
    const { assetTypeList, keyField, model } = this.state;
    const { actions } = this.props;
    let data = {};
    for (var key in assetTypeList) {
      if (curId === assetTypeList[key].name) {
        data = assetTypeList[key];
      }
    }

    actions.overlayerShow(<TypeEditPopup id="updateType"  title={this.formatIntl('equipment.addDevice')}
      data={data}
      idEdit={false}
      hasPower={true}
      hasLife={true}
      hasManufacture={true}
      onCancel={() => {
        actions.overlayerHide();
      }}
      onConfirm={(data) => {
        //data={type, power, serviceLife, manufacturer, detail,...others}
        //根据data中的参数更新设备型号，并将结果传入API实现数据的更改
        let name = data.name;
        let power = data.power;
        let description = data.description;
        let life = data.life;
        let manufacture = data.manufacture;
        let typeData = Object.assign({}, {
          name: name, power: power, description: description,
          life: life, manufacture: manufacture
        });
        //对更新后的数据做合并处理，名字是主键不能更改
        let list = assetTypeList;
        list.pop();
        let editData = list.map(item => {
          return item.name === typeData.name ? typeData : item
        })
        actions.overlayerHide();
        updateModelTypeByModel(model, editData, (data) => {
          this.mounted && this.getModelType();
        })
      }}></TypeEditPopup>);
  }

  rowAdd() { //添加设备状态，不传入Id，
    let curId = '';
    let { assetTypeList, model } = this.state;
    const { actions } = this.props;
    let data = {};
    data.name = '';
    data.description = '';
    data.power = '';
    data.life = '';
    data.manufacture = '';
    actions.overlayerShow(<TypeEditPopup id="updateType" title={this.formatIntl('equipment.addDevice')}
      addNotify={actions.addNotify}
      data={data}
      hasPower={true}
      hasLife={true}
      hasManufacture={true}
      onCancel={() => {
        actions.overlayerHide();
      }}
      onConfirm={(data) => {
        //data={type, power, serviceLife, manufacturer, detail, ...others}
        //根据data中的参数更新设备型号，并将结果传入API实现数据的更改
        let name = data.name;
        let power = data.power;
        let life = data.life;
        let manufacture = data.manufacture;
        let description = data.description;
        let typeData = Object.assign({}, {
          name: name, description: description
        });
        assetTypeList.pop()
        let dataAdd = assetTypeList;
        dataAdd.push(typeData);
        let resPram = ''
        updateModelTypeByModel(model, dataAdd, (data) => {
          this.mounted && this.getModelType();
          actions.overlayerHide();
        }, resPram, (msg) => {
          actions.addNotify(0, msg.message)
        })
      }}></TypeEditPopup>);
  }



  render() {
    const { data, assetPropertyList, assetTypeList, keyField } = this.state;
    let length = assetTypeList.length;
    return (
      <Content>
        <div className="row heading">
          <div className="type"><span></span>{this.formatIntl('asset.assetTypes')}</div>
          <table className="equipment">
            <thead>
              <tr>
                {
                  this.assetTypeColumns.map((column, index) => {
                    return <th key={index}>{column.title}</th>;
                  })
                }
                {<th>{this.formatIntl('equipment.edit')}</th>}
              </tr>
            </thead>
            <tbody>
              {
                assetTypeList.map((row, index) => {
                  return <tr key={index}>
                    {
                      this.assetTypeColumns.map((column, index) => {
                        return <td key={index}>{row[column.field]}</td>;
                      })
                    }
                    {index !== length - 1 ?
                      <td className="edit">
                        <a className="btn" role="presentation">
                          <span onClick={() => { this.rowEdit(row[keyField]); }} role="presentation"
                            className="icon_edit">{this.formatIntl('equipment.modify')}</span>
                        </a>
                        <a className="btn" role="presentation">
                          <span onClick={() => { this.rowDelete(row[keyField]); }} role="presentation"
                            className="icon_delete">{this.formatIntl('equipment.delete')}</span></a>
                      </td>
                      :
                      <td><span role="presentation" className="btn"
                        onClick={this.rowAdd}>+{this.formatIntl('equipment.add')}</span> </td>}
                  </tr>;
                })
              }
            </tbody>
          </table>
        </div>
      </Content>
    );
  }
}


function mapStateToProps(state) {
  return {
    sidebarNode: state.assetManage.get('sidebarNode'),
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      addNotify: addNotify,
      overlayerShow: overlayerShow,
      overlayerHide: overlayerHide,
    }, dispatch),
  };
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(injectIntl(Gateway));