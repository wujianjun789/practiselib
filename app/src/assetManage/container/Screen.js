/**
 * Created by a on 2017/8/23.
 */
import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import Content from '../../components/Content';

import {getModelData, getModelProps, getModelTypes, getModelDefaultsValues, getModelDefaults} from '../../data/assetModels';
import Immutable from 'immutable';

import {FormattedMessage,injectIntl} from 'react-intl';
import { intlFormat } from '../../util/index';

export class Screen extends Component {
    constructor(props) {
        super(props);
        this.state = {
            model:'screen',
            devicePro:Immutable.fromJS([
                '软件版本',
                '系统版本',
                '内核版本'
            ]),
            data:Immutable.fromJS([
                {type:'LC300', detail:'LC300灯控'},
                {type:'LC600', detail:'LC600灯控'},
                {type:'LCMINI', detail:'智慧路灯用'}
            ]),
            dataDefaults: [],
            dataDefaultsValues: Immutable.fromJS([]),
        };

        // this.columns = [{field:"type", title:"型号"}, {field:"detail", title:"描述"}]
        this.columns = [{field:'type', title:intlFormat({en:'type',zh:'型号'})}, {field:'detail', title:intlFormat({en:'detail',zh:'描述'})}]; 

        this.initTreeData = this.initTreeData.bind(this);
    }

    componentWillMount(){
        this.mounted = true;
        getModelData(()=>{this.mounted && this.initTreeData();});
    }

    componentWillUnmount(){
        this.mounted = false;
    }

    initTreeData(){
        const {model} = this.state;
        this.setState({
            devicePro:Immutable.fromJS(getModelProps(model)),
            data: Immutable.fromJS(getModelTypes(model)),
            dataDefaults: getModelDefaults(model),
            dataDefaultsValues: Immutable.fromJS(getModelDefaultsValues(model)),
        });
    }

    render() {
        const { data, devicePro, dataDefaults, dataDefaultsValues } = this.state;
        return (
            <Content>
                <div className="row heading">
                    <div className="property"><span></span>{this.props.intl.formatMessage({id:'asset.property'})}</div>
                    <ul className="property-list">
                        {
                            devicePro.map((item,index)=>{
                                return <li key={index}>{item}</li>;
                            })
                        }
                    </ul>
                </div>
                <div className="row heading">
                    <div className="type"><span></span>{this.props.intl.formatMessage({id:'asset.type'})}</div>
                    <table className="equipment">
                        <thead>
                            <tr>
                                {
                                    this.columns.map((column,index)=>{
                                        return <th key={index}>{column.title}</th>;
                                    })
                                }
                            </tr>
                        </thead>
                        <tbody>
                            {
                                data.map((row, index)=>{
                                    return <tr key={index}>
                                        {
                                            this.columns.map((column,index)=>{
                                                return <td key={index}>{row.get(column.field)}</td>;
                                            })
                                        }
                                    </tr>;
                                })
                            }
                        </tbody>
                    </table>
                </div>
                <div className="row heading">
                    <div className="param"><span></span>{this.props.intl.formatMessage({id:'asset.param'})}</div>
                    <table className="equipment">
                        <thead>
                            <tr>
                                {
                                    this.columns.map((column,index)=>{
                                        return <th key={index}>{column.title}</th>;
                                    })
                                }
                            </tr>
                        </thead>
                        <tbody>
                            {
                                dataDefaultsValues.map((row, index)=>{
                                    return <tr key={index}>
                                        {
                                            this.columns.map((column,index)=>{
                                                return <td key={index}>{row.get(column.field)}</td>;
                                            })
                                        }
                                    </tr>;
                                })
                            }
                        </tbody>
                    </table>
                </div>
            </Content>
        );
    }
}


function mapStateToProps(state) {
    return {
        sidebarNode: state.assetManage.get('sidebarNode')
    };
}

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
        }, dispatch)
    };
}
export default connect(
    mapStateToProps,
    mapDispatchToProps
)(injectIntl(Screen));