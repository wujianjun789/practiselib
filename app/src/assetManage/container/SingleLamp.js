/**
 * Created by a on 2017/8/23.
 */
import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import Content from '../../components/Content'

import {getModelData, getModelProps, getModelTypes, getModelDefaultsValues, getModelDefaults} from '../../data/assetModels'
import Immutable from 'immutable';

export class SingleLamp extends Component {
    constructor(props) {
        super(props);
        this.state = {
            model:"lc",
            devicePro:Immutable.fromJS([
                "软件版本",
                "系统版本",
                "内核版本"
            ]),
            data:Immutable.fromJS([
                {type:"LC300", detail:"LC300灯控"},
                {type:"LC600", detail:"LC600灯控"},
                {type:"LCMINI", detail:"智慧路灯用"}
            ]),
            dataDefaults: [],
            dataDefaultsValues: Immutable.fromJS([]),
        }

        this.columns = [{field:"type", title:"型号"}, {field:"detail", title:"描述"}]

        this.initTreeData = this.initTreeData.bind(this);
    }

    componentWillMount(){
        this.mounted = true;
        getModelData(()=>{this.mounted && this.initTreeData()});
    }

    componentWillUnmount(){
        this.mounted = false;
    }

    initTreeData(){
        const {model} = this.state;
        this.setState({
            devicePro:Immutable.fromJS(getModelProps(model)),
            data: Immutable.fromJS(getModelTypes(model)),
            dataDefaults: getModelDefaults(model),
            dataDefaultsValues: Immutable.fromJS(getModelDefaultsValues(model)),
        })
    }

    render() {
        const { data, devicePro, dataDefaults, dataDefaultsValues } = this.state
        return (
            <Content>
                <div className="row heading">
                    <div className="property"><span></span>设备属性</div>
                    <ul className="property-list">
                        {
                            devicePro.map((item,index)=>{
                                return <li key={index}>{item}</li>
                            })
                        }
                    </ul>
                </div>
                <div className="row heading">
                    <div className="type"><span></span>设备类别</div>
                    <table className="equipment">
                        <thead>
                        <tr>
                            {
                                this.columns.map((column,index)=>{
                                    return <th key={index}>{column.title}</th>
                                })
                            }
                        </tr>
                        </thead>
                        <tbody>
                        {
                            data.map((row, index)=>{
                                return <tr key={index}>
                                    {
                                        this.columns.map((column,index)=>{
                                            return <td key={index}>{row.get(column.field)}</td>
                                        })
                                    }
                                </tr>
                            })
                        }
                        </tbody>
                    </table>
                </div>
                <div className="row heading">
                    <div className="param"><span></span>默认参数</div>
                    <table className="equipment">
                        <thead>
                        <tr>
                            {
                                dataDefaults.map((column,index)=>{
                                    return <th key={index}>{column.title}</th>
                                })
                            }
                        </tr>
                        </thead>
                        <tbody>
                        {
                            dataDefaultsValues.map((row, index)=>{
                                return <tr key={index}>
                                    {
                                        dataDefaults.map((column,index)=>{
                                            return <td key={index}>{row.get(column.field)}</td>
                                        })
                                    }
                                </tr>
                            })
                        }
                        </tbody>
                    </table>
                </div>
            </Content>
        )
    }
}


function mapStateToProps(state) {
    return {
        sidebarNode: state.assetManage.get('sidebarNode')
    }
}

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
        }, dispatch)
    }
}
export default connect(
    mapStateToProps,
    mapDispatchToProps
)(SingleLamp);