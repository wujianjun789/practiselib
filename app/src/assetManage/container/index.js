/**
 * Created by a on 2017/7/17.
 */
import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import {injectIntl} from 'react-intl';

import '../../../public/styles/assetManage-model.less';
import '../../../public/styles/assetManage-statistics.less';

import HeadBar from '../../components/HeadBar';
import SideBar from '../../components/SideBar';
import Overlayer from '../../common/containers/Overlayer';

import {getModelData, TreeData} from '../../data/assetModels';
import {treeViewInit, onToggleById} from '../../common/actions/treeView';

import {sideBarToggled} from '../action/index';
import {treeViewNavigator} from '../../common/util/index';
class AssetManageIndex extends Component {
  constructor(props) {
    super(props);

    this.state = {
      pathname: ""
    }

    this.initModel = this.initModel.bind(this);
    this.initTreeData = this.initTreeData.bind(this);
    this.onToggle = this.onToggle.bind(this);

    this.getMathNode = this.getMathNode.bind(this);
  }

  componentWillMount() {
    this.mounted = true;
    this.initModel();
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  componentWillReceiveProps(nextProps){
    if(nextProps.location.pathname !==  this.props.location.pathname){
      this.props.actions.onToggleById(nextProps.location.pathname);
    }
  }
  componentDidUpdate(){
    // console.log(this.props);
    // const {location} = this.props;
    // if(location.pathname !== this.state.pathname){
    //   this.setState({pathname: location.pathname}, ()=>{
    //     this.initModel();
    //   });
    // }
  }
  initModel(){
    getModelData(() => {this.mounted && this.initTreeData();});
  }

  initTreeData() {
    TreeData.map(tree=>{
      return Object.assign({}, tree, {name:this.props.intl.formatMessage({id:tree.name})});
    })
    this.props.actions.treeViewInit(TreeData);
    treeViewNavigator(TreeData, this.props.router);

    const node = this.getMathNode(TreeData, this.props.location.pathname);
    if(node){
      this.onToggle(node);
    }
  }

  getMathNode(data, pathname){
    for(let i=0;i<data.length;i++){
      if(!data[i].children && data[i].link.indexOf(pathname) > -1){
        return data[i];
      }else{
        if(data[i].children){
          const node = this.getMathNode(data[i].children, pathname);
          if(node){
            return node;
          }
        }

      }
    }

    return null;
  }

  onToggle(node) {
    const {actions} = this.props;
    actions.sideBarToggled(node);
  }

  render() {
    let parentPath = '';
    let childPath = '';
    const {routes} = this.props;
    if (routes.length > 4) {
      parentPath = routes[4].path;
    }

    if (routes.length > 5) {
      childPath = routes[5].path;
    }

    return <div className={'container ' + 'asset-' + parentPath + ' ' + parentPath + '-' + childPath}>
      <HeadBar moduleName="app.asset.manage" router={this.props.router}/>
      <SideBar onToggle={this.onToggle}/>
      {this.props.children}
      <Overlayer />
    </div>;
  }
}

function mapStateToProps(state) {
  return {
    userCenter:state.userCenter,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      treeViewInit: treeViewInit,
      onToggleById: onToggleById,
      sideBarToggled: sideBarToggled,
    }, dispatch),
  };
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(injectIntl(AssetManageIndex));