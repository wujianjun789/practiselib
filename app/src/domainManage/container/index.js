/**
 * Created by a on 2017/7/17.
 */
import React,{Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import HeadBar from '../../components/HeadBar'
import SideBar from '../../components/SideBar'
import Overlayer from '../../common/containers/Overlayer'

import {TreeData} from '.././../data/domainModel'
import {treeViewInit} from '../../common/actions/treeView'
import {sideBarToggled} from '../action/index'
class DomainManageIndex extends Component{
    constructor(props){
        super(props);
        this.initTreeData = this.initTreeData.bind(this);
        this.onToggle = this.onToggle.bind(this);
    }

    componentWillMount(){
        if(sessionStorage.sessionID==0||sessionStorage.sessionID==null||sessionStorage.sessionID==""){
            this.props.router.push('/login') 
        }else{}
        this.initTreeData();
    }

    componentWillUnmount(){

    }

    componentDidMount(){
        console.log(this.props.userCenter); 
    }

    initTreeData(){
        this.props.actions.treeViewInit(TreeData);
    }

    onToggle(node){
        this.props.actions.sideBarToggled(node);
    }

    render(){
        let path = "";
        const {children} = this.props;
        if(children){
            path = children.props.route.path;
        }
        return <div className={"container "+"domain-"+path}>
            <HeadBar moduleName={/*path=="domainEdit"?*/"域管理"/*:"地图预览"*/} router={this.props.router}/>
            <SideBar onToggle={this.onToggle}/>
            {this.props.children}
            <Overlayer />
        </div>
    }
}

function mapStateToProps(state) {
    return {
        userCenter:state.userCenter
    }
}

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            treeViewInit: treeViewInit,
            sideBarToggled: sideBarToggled
        }, dispatch)
    }
}
export default connect(
    mapStateToProps,
    mapDispatchToProps
)(DomainManageIndex);