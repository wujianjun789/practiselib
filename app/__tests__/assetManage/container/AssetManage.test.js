/**
 * Created by a on 2017/7/21.
 */
import React from 'react';
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer'

import {SingleLamp} from '../../../src/assetManage/container/SingleLamp';
import {Gateway} from '../../../src/assetManage/container/Gateway';
import {Pole} from '../../../src/assetManage/container/Pole';
import {Screen} from '../../../src/assetManage/container/Screen';
import {Sensor} from '../../../src/assetManage/container/Sensor';
import {Xes} from '../../../src/assetManage/container/Xes';

test('AssetManage renders', ()=>{
    const component = renderer.create(
        <SingleLamp />
    )

    let assetManage = component.toJSON();
    expect(assetManage).toMatchSnapshot();
})

test('AssetManage div click', ()=>{
    const component = shallow(<SingleLamp />)
    expect(component.find('.property').length).toEqual(1);
    expect(component.find('.type').length).toEqual(1);
})
