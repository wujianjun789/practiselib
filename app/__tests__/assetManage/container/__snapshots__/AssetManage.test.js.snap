exports[`test AssetManage renders 1`] = `
<div
  className="content undefined">
  <div
    className="row heading">
    <div
      className="property">
      <span />
      设备属性
    </div>
    <ul
      className="property-list">
      <li>
        软件版本
      </li>
      <li>
        系统版本
      </li>
      <li>
        内核版本
      </li>
    </ul>
  </div>
  <div
    className="row heading">
    <div
      className="type">
      <span />
      设备类别
    </div>
    <table
      className="equipment">
      <thead>
        <tr>
          <th>
            型号
          </th>
          <th>
            描述
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            LC300
          </td>
          <td>
            LC300灯控
          </td>
        </tr>
        <tr>
          <td>
            LC600
          </td>
          <td>
            LC600灯控
          </td>
        </tr>
        <tr>
          <td>
            LCMINI
          </td>
          <td>
            智慧路灯用
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
